BBBBroadcast/SampleHandler.swift

```diff
@@ -0,0 +1,149 @@
+        guard let javascriptData = defaults?.object(forKey: Constants.javascriptData) as? Data else { return }
+        do {
+            let jsData = try JSONDecoder().decode(JavascriptData.self, from: javascriptData)
+            javascriptPayload = jsData.payload
+            setupWebSocketConnection(with: jsData.websocketUrl)
+        } catch (let error) {
+            print("⚡️☠️ Failed to load payload data: \(error.localizedDescription)")
+        }
+    }
+    
+    override func broadcastPaused() {
+        // User has requested to pause the broadcast. Samples will stop being delivered.
+        print("Broadcast has paused.")
+    }
+    
+    override func broadcastResumed() {
+        // User has requested to resume the broadcast. Samples delivery will resume.
+        print("Broadcast has resumed.")
+    }
+    
+    override func broadcastFinished() {
+        // User has requested to finish the broadcast.
+        print("Broadcast has finished.")
+    }
+    
+    override func processSampleBuffer(_ sampleBuffer: CMSampleBuffer, with sampleBufferType: RPSampleBufferType) {
+        print("Broadcast sample buffer received")
+        switch sampleBufferType {
+        case .video:
+            // Handle video sample buffer
+            guard isConnected, let imageBuffer: CVImageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer) else { break }
+//            let pixelFormat = CVPixelBufferGetPixelFormatType(imageBuffer) // kCVPixelFormatType_420YpCbCr8BiPlanarFullRange
+            let timeStampNs: Int64 = Int64(CMTimeGetSeconds(CMSampleBufferGetPresentationTimeStamp(sampleBuffer)) * 1000000000)
+            let rtcPixlBuffer = RTCCVPixelBuffer(pixelBuffer: imageBuffer)
+            let rtcVideoFrame = RTCVideoFrame(buffer: rtcPixlBuffer, rotation: ._0, timeStampNs: timeStampNs)
+            webRTCClient.push(videoFrame: rtcVideoFrame)
+            break
+        case .audioApp:
+            // Handle audio sample buffer for app audio
+            break
+        case .audioMic:
+            // Handle audio sample buffer for mic audio
+            break
+        @unknown default:
+            // Handle other sample buffer types
+            fatalError("Unknown type of sample buffer")
+        }
+    }
+    
+    // MARK: Setup WebSocket
+
+    private func setupWebSocketConnection(with url: String) {
+        guard let websocketUrl = URL(string: url) else { return }
+        // Use Native socket to establish connection
+        let websocketProvider: WebSocketProvider = NativeWebSocket(url: websocketUrl)
+        signalingClient = SignalingClient(webSocket: websocketProvider)
+        signalingClient?.delegate = self
+        signalingClient?.connect()
+    }
+    
+    // MARK: - WebRTC
+
+    private func sendInitialSocketMessageWithSdpOffer() {
+        webRTCClient.offer { [weak self] (localSdpOffer) in
+            guard let `self` = self, var data = self.javascriptPayload else { return }
```

Comment:
How about a different variable name instead of the backticks?

Reply:
Well we can use strongSelf, but Im more for this kind of unwrapping self because you can use it later in code just as self.